### ISSUE TITLE
Package update button indicates "Package updates are available" even when that statement is false

### ISSUE THREAD

**Issue by cjsha** [bonsai-rx/bonsai#2003]
This button indicates package updates are available even when there aren't actually any package updates available. 

![Screenshot 2024-09-06 110523](https://github.com/user-attachments/assets/7c2624bd-aa13-47e8-b7c6-9413094f9721)

I suppose it should be greyed out with a different tooltip i.e. "No package updates available" when there are no package updates available?

I like the idea behind this feature a lot btw.

**Comment by cjsha** [comment]
I'm sorry, I didn't realize 2.9 feedback should go in the discussions, you explicitly mention that in the 2.9 pre-release announcement. I'm going to close this and make it a comment in the discussion.

**Comment by glopesdev** [comment]
@cjsha Thanks for the feedback, this is actually probably an issue related to package type filtering, I am reopening the issue and will investigate.

> I suppose it should be greyed out with a different tooltip i.e. "No package updates available" when there are no package updates available?

This is also a great idea and probably better than the current hiding of the button which makes it harder to discover (this feature has actually been there for a long time).

**Comment by cjsha** [comment]
ah you're right. I needed to check that "Show Dependencies" box for the available updates to appear. I'm not accustomed to that checkbox. 

But now that I've updated, Bonsai crashes when I open the workflow editor in the portable Bonsai 2.9 environment. The only packages I had previously installed was the starter pack. 

Here is a gif of me updating packages and the workflow editor crashing afterwards.

![bonsai-package-update](https://github.com/user-attachments/assets/95b59104-c214-49ad-8f5d-99427726e755)

Here is a gif of me trying to open the workflow editor thereafter. I can confirm that the workflow editor was opening prior to downloading those updates.

![bonsai-crash-due-to-update](https://github.com/user-attachments/assets/8b88c468-9583-4870-85d1-28aaacb8daf4)

I'm not sure what the best way to provide troubleshooting details, if you want them, but it seems repeatable on my machine.

**Comment by glopesdev** [comment]
@cjsha yeah, this is probably some dependency that breaks, we need to be much more careful with constraining updates if we are to embrace NuGet fully.

Thanks for the report, will investigate.
